src/nyc_trees/apps/home/views.py

```diff
@@ -21,6 +24,28 @@ def retrieve_job_status(request):
     pass
 
 
+def redirect_to_flat_page(url):
+    def fn(request, *args, **kwargs):
+        return flatpage(request, url)
+    return fn
+
+
+def mark_user_on_success(attr):
+    def outer_decorator(view_fn):
+        @wraps(view_fn)
+        def wrapper(request, *args, **kwargs):
+            # do this first to make sure the wrapped view doesn't
+            # raise
+            ctx = view_fn(request, *args, **kwargs)
+            user = request.user
+            if user.is_authenticated() and not getattr(user, attr):
+                setattr(user, attr, True)
+                user.save()
+            return ctx
+        return wrapper
+    return outer_decorator
+
+
```

Comment:
Neither of these functions are really views.  Consider moving both of these into a new `home/decorators.py` file.